data_diff/parse_time.py

```diff
@@ -0,0 +1,61 @@
+import re
+from datetime import datetime, timedelta
+from difflib import SequenceMatcher
+
+TIME_UNITS = dict(
+    seconds="seconds",
+    minutes="minutes",
+    hours="hours",
+    days="days",
+    weeks="weeks",
+    months="months",
+    years="years",
+    # Shortcuts
+    s="seconds",
+    min="minutes",
+    h="hours",
+    d="days",
+    w="weeks",
+    mon="months",
+    y="years",
+)
+
+EXTRAPOLATED = {"months": (30, "days"), "years": (365, "days")}
+assert set(EXTRAPOLATED) <= set(TIME_UNITS)
+
+TIME_RE = re.compile(r"(\d+)([a-z]+)")
+
+
+def string_similarity(a, b):
+    return SequenceMatcher(None, a, b).ratio()
+
+
+def parse_time_atom(count, unit):
+    count = int(count)
+    try:
+        unit = TIME_UNITS[unit]
+    except KeyError:
+        most_similar = max(TIME_UNITS, key=lambda k: string_similarity(k, unit))
+        raise ValueError(f"'{unit}' is not a recognized time unit. Did you mean '{most_similar}'?")
+
+    if unit in EXTRAPOLATED:
+        mul, unit = EXTRAPOLATED[unit]
+        count *= mul
+    return count, unit
+
+
+def parse_time_delta(t: str):

```

Comment:
What if I pass `1h30min`? I think it'll just interpret it as `1h`, even though that's not what I meant. Probably worth erroring in this case?

Reply:
What about floats? E.g. `1.5h` isn't crazy

We should probably support either floats or using multiple deltas like `1h30min`.

Reply:
We already support `1h30min` ;)

Reply:
I missed the while! 👍🏻 👍🏻